scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+         wsrep_log_info "DONOR SAY SST ($RP_PURGED)"
+         readonly CLEANUP_CLONE_PLUGIN=${RP_PURGED%|SST*}
+         RP_PURGED_EMERGENCY=${RP_PURGED#*SST@}
+         RP_PURGED_EMERGENCY=${RP_PURGED_EMERGENCY%"<EOF>"}
+         wsrep_log_debug "-> CLONE Plugin to remove? (yes/no) = $CLEANUP_CLONE_PLUGIN"
+         wsrep_log_debug "-> recovered position from DONOR: $RP_PURGED_EMERGENCY"
+    fi
+
+    ##################################################################################################
+    # If we need to SST in any case we must remove the data, so let us do it here and be sure we work on a clean directory
+    wsrep_log_info "Cleaning Data directory $WSREP_SST_OPT_DATA"
+    rm -fr $WSREP_SST_OPT_DATA/*
+
+    # Before starting let us be sure we remove Netcat given it is using same MySQL port
+    # check if there is another netcat process on the IP port we need and try to kill it.
+    NETCAT_KILL=`ps -xo pid,command|grep -e "nc -l -k $JOINER_CLONE_HOST $JOINER_CLONE_PORT"|grep -v "grep"|cut -d ' ' -f 1`
+    if [ ! "$NETCAT_KILL" == "" ];then
+        wsrep_log_info "-> NetCat still up PID: $NETCAT_KILL. Will try to kill it"
+        kill -9 $NETCAT_KILL
+        if [ "$?" != "0" ]; then
+            wsrep_log_debug "-> Cannot kill the existing NetCat process PID $NETCAT_KILL"
+        fi
+    fi
+
+    NC_PID=""
+
+    # No data dir, need to initialize one first, to make connections to
+    # this node possible.
+    # We need to use a temporary empty data directory, because the
+    # actual datadir may already contain some wsrep-related files
+    wsrep_log_info "Initializing data directory at $tmp_datadir"
+
+    wsrep_log_info "INITIALIZE DB"
+
+    echo "" > $CLONE_ERR
+    eval $CLONE_ENV $CLONE_BINARY $DEFAULT_OPTIONS \
+         $SKIP_NETWORKING_OPTIONS \
+         --initialize-insecure --wsrep_provider=none --datadir="$tmp_datadir" >> $CLONE_ERR 2>&1 || \
+    ( wsrep_log_error "Failed to initialize data directory. Some hints below:"
+      grep '[ERROR]' $CLONE_ERR | cut -d ']' -f 4- | while read msg
+      do
+          wsrep_log_error "> $msg"
+      done
+      wsrep_log_error "Full log at $CLONE_ERR"
+      exit 1 )
+
+    # Move initialized data directory structure to real datadir and cleanup
+    mv --force "$tmp_datadir"/* "$WSREP_SST_OPT_DATA/"
+    sleep 2
+    wsrep_log_debug "-> REMOVE $tmp_datadir"
+    rm -rf "$tmp_datadir"
+
+    wsrep_log_info "INITIALIZE DB DONE"
+
+    # Need to create an extra user for 'localhost' because in some installations
+    # by default exists user ''@'localhost' and shadows every user with '%'.
+ cat << EOF > "$CLONE_SQL"
+ SET SESSION sql_log_bin=OFF;
+ CREATE USER '$CLONE_USER'@'%' IDENTIFIED $WITH_OPTION BY '$CLONE_PSWD';
+ GRANT CLONE_ADMIN, SYSTEM_VARIABLES_ADMIN, SUPER, SHUTDOWN, EXECUTE ON *.* to '$CLONE_USER'@'%';
+ GRANT INSERT ON mysql.plugin to '$CLONE_USER';
+ GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER';
```

Comment:
Probably we need the host `%` as well
```suggestion
 GRANT INSERT ON mysql.plugin to '$CLONE_USER'@'%';
 GRANT SELECT,UPDATE,INSERT ON performance_schema.* TO '$CLONE_USER'@'%';
```

Reply:
if omitted it defaulted to '%' anyhow. We can add it to be consistent and verbose, will not change the behaviour.

Reply:
done